code/core/bin/loader.mjs

```diff
@@ -0,0 +1,33 @@
+import { readFileSync } from 'node:fs';
+import { fileURLToPath } from 'node:url';
+
+import { transformSync } from 'esbuild';
+
+export function load(url, context, nextLoad) {
+  /** Convert TS to ESM using esbuild */
+  if (
+    url.endsWith('.ts') ||
+    url.endsWith('.tsx') ||
+    url.endsWith('.mts') ||
+    url.endsWith('.cts') ||
+    url.endsWith('.tsx') ||
+    url.endsWith('.mtsx') ||
+    url.endsWith('.ctsx')
+  ) {
+    const rawSource = readFileSync(fileURLToPath(url), 'utf-8');
+    const transformedSource = transformSync(rawSource.toString(), {
+      loader: 'ts',
+      target: 'es2020',
```

Comment:
I think we should instead import the `NODE_TARGET` variable we're also using for our own bundling.